### PR TITLE
pathogen-repo-build: Move `build.log` to ignored directory

### LINKED ISSUE #77: CI: pathogen-repo-build (aws-batch) does not output full logs to build.log
## Context
First noticed when verifying tests for https://github.com/nextstrain/.github/pull/76. 

The CI workflow test for the `pathogen-repo-build` workflow using the `aws-batch` runtime does not output full logs to `build.log`. Most importantly, the line containing `AWS Batch Job ID` is not included in the `build.log` file so the workflow is unable to use the `AWS_BATCH_JOB_ID` to create the AWS Batch job summary. 

You can see the [job id in the logs](https://github.com/nextstrain/.github/actions/runs/8210709255/job/22458631526#step:8:40) via the GitHub Action UI. However, the workflow sets an [empty `AWS_BATCH_JOB_ID` envvar](https://github.com/nextstrain/.github/actions/runs/8210709255/job/22458631526?pr=76#step:9:14) and the `build.log` file of the [artifact](https://github.com/nextstrain/.github/actions/runs/8210709255/artifacts/1311002405) is also truncated.

---
## Possible solution

1. I tried using `tee -a build.log` in https://github.com/nextstrain/.github/commit/a8d9f4eccbfde5874a41c5ddd8336062a7abe1f1, but the `build.log` file of the [artifact](https://github.com/nextstrain/.github/actions/runs/8209746229/artifacts/1310798412) was still truncated. 

2. I thought maybe setting `PYTHONUNBUFFERED` in https://github.com/nextstrain/.github/commit/6f69e717b75f6e4c44e494cfe37bc104c4315661 would help, but the workflow is now using the standalone CLI so no dice. 

There must be some sort of stdout buffering that is getting lost and not output to the `build.log` file, but I'm out of ideas for now.

### PR DESCRIPTION
**Pull request by joverlee521**
## Description of proposed changes

See commit for details.

## Related issue(s)

Resolves #77 

## Checklist

- [x] Checks pass

### CODE REVIEW

**Comment by joverlee521** [comment]
The CI workflow included the [AWS Batch job summary](https://github.com/nextstrain/.github/actions/runs/8302225248/attempts/1#summary-22724088158) 🎉 

The CI workflow now reveals a separate issue...In the CI workflow, the aws-batch build is run **_without_** the `--detach` flag and is able to complete the build within the `run-build` job. However, the `wait-1` job still re-attaches to the AWS Batch job because it's only run condition is [the existence of the `AWS_BATCH_JOB_ID`](https://github.com/nextstrain/.github/blob/5cd0ba1f61c073f8ecdb6f71d25444b337cdc4eb/.github/workflows/pathogen-repo-build.yaml.in#L283). This is not a big deal since the `wait-1` job will only take a couple seconds to download a completed build and our automated workflows usually use the `--detach flag`. 

I don't think it's worth complicating the workflow logic just to prevent an extra `wait-1` job, but please chime in if there's a good reason to update it!

File: .github/workflows/pathogen-repo-build.yaml [hunk 1]
```diff
@@ -274,14 +275,14 @@ jobs:
           # shellcheck disable=SC2154
           set -x
 
-          eval "$NEXTSTRAIN_BUILD_COMMAND" |& tee build.log
+          eval "$NEXTSTRAIN_BUILD_COMMAND" |& tee "$NEXTSTRAIN_BUILD_LOG"
       # Attempt to get the AWS Batch ID even if the run build command failed
```
Comment (victorlin) [inline .github/workflows/pathogen-repo-build.yaml:279]: Suggestion: `tee` to `.git/build.log` then move it to `build.log` after `nextstrain build` is finished. This allows presence under `.git` to be kept as an implementation detail.

Comment (joverlee521) [reply .github/workflows/pathogen-repo-build.yaml:285]: OH! Good idea! Added in https://github.com/nextstrain/.github/pull/78/commits/b9a9ae9b2aa12a5943834ea387e322db08788a73

**Comment by joverlee521** [comment]
Merging since I intend to do some additional work on the pathogen-repo-build workflow to [support role assumption in other runtimes](https://github.com/nextstrain/.github/pull/53#discussion_r1462484764).

**Pull request merged**